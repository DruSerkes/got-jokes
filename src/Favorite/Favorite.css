li {
    list-style-type: none;
    position: relative;
    padding: 2rem;
}

.Favorite-removeButton {
    position: absolute;
    top: 0;
    right: 0;
    padding: 0;
}

.Favorite-removeButton:hover {
    color: red !important;
}