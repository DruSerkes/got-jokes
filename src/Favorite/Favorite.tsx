import * as React from 'react';
import { jokeData } from '../types';
import './Favorite.css';

interface FavoritesProps {
    fav: jokeData;
}

export const Favorite = ({ fav }: FavoritesProps) => {
    const generateFav = (joke: jokeData) => {
        if (joke.type === 'twopart') {
            return (
                <li className='py-3 list-group-item-dark border-bottom rounded border-dark'>
                    {joke?.setup}
                    <br />
                    {joke?.delivery}
                </li>
            );
        } else {
            return (
                <li className='py-3 list-group-item-dark border-bottom rounded border-dark'>
                    {joke?.joke}
                </li>
            );
        };
    };

    return generateFav(fav);
}