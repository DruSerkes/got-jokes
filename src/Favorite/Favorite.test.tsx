import * as React from 'react';
import { render } from '@testing-library/react';
import { Favorite } from './Favorite';



describe('Favorites tests', () => {
    const testJoke = {
        "error": false,
        "category": 'dark',
        "type": 'single',
        "joke": 'test joke',
        "flags": {
            "nsfw": false,
            "religious": false,
            "political": false,
            "racist": false,
            "sexist": false,
            "explicit": false
        },
        "id": 13,
        "safe": true,
        "lang": 'en',
        'setup': 'test setup',
        'delivery': 'test delivery'
    };

    it('should render and match snapshot', () => {
        const { asFragment } = render(<Favorite fav={testJoke} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should show a joke', () => {
        const { getByText } = render(<Favorite fav={testJoke} />);
        expect(getByText(testJoke.joke)).toBeInTheDocument();
    });
})