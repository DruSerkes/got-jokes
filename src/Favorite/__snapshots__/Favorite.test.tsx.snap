// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites tests should render and match snapshot 1`] = `
<DocumentFragment>
  <li
    class="py-3 list-group-item-dark border-bottom rounded border-dark"
  >
    test joke
  </li>
</DocumentFragment>
`;
